src/SponsorshipLib.sol

```diff
@@ -0,0 +1,134 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.20;
+
+import {IERC20Minimal} from "./Interfaces.sol";
+
+library SponsorshipLib {
+    
+    struct SponsorProfile { 
+        string name; 
+        string image; 
+    }
+
+    struct TokenSubmission {
+        string name;
+        address submitter;
+        bool exists;
+    }
+
+    struct SponsorshipStorage {
+        mapping(address => mapping(address => uint256)) sponsorTokenAmounts; // sponsor => token => amount
+        mapping(address => uint256) approvedTokensAmount;                    // token => total approved amount
+        mapping(address => uint256) tokenMinAmount;                          // token => minimum amount per deposit
+        mapping(address => SponsorProfile) sponsorProfiles;                  // sponsor => profile
```

Comment:
I suggest moving everything related to profiles to a separate `Profile` contract. In the hackathon/sponsorship contract/lib, we would keep only the address of the sponsor. Any other detail would be stored in the `Profile` contract. I see the following advantages:

1 - It could help with the size limitations that you are facing.

2 - The profiles could then be reused across hackathons and even in other projects of ours.

3 - This Profile contract could eventually become its own project, serving a role similar to ENS.